client/app/components/ui/question.hbs

```diff
@@ -0,0 +1,18 @@
+{{#let
+  (concat 'ui/' (if @group 'fieldset-legend' 'label-text'))
+  as |promptComponent|
+}}
+  {{#if @group}}
+    <fieldset class="medium-margin-bottom" ...attributes>
+      {{yield (hash
+        prompt=(component promptComponent)
+      )}}
+    </fieldset>
+  {{else}}
+    <label class="medium-margin-bottom" ...attributes>
+      {{yield (hash
+        prompt=(component promptComponent)
+      )}}
+    </label>
+  {{/if}}
+{{/let}}
```

Comment:
I'd simplify this a bit more:

```handlebars
  {{#if @group}}
    <fieldset class="medium-margin-bottom" ...attributes>
      {{yield (hash
        Prompt=(component 'ui/fieldset-legend')
      )}}
    </fieldset>
  {{else}}
    <label class="medium-margin-bottom" ...attributes>
      {{yield (hash
        Prompt=(component 'ui/label-text')
      )}}
    </label>
  {{/if}}
```

Reply:
Duh! I totally overcomplicated this because I arrived at it from a different angle where I needed to dynamically bind some args. Thanks for catching that.